### PR TITLE
Fix compatible issue and refine null parameters checking 

### PR DESCRIPTION
**Pull request by dvdli**
For the compatibility with version 1.x, do pcm_prepare before starting the pcm driver.

### CODE REVIEW

**Comment by dvdli** [comment]
@tay10r @simonrules @gkasten Could you please help to review this patch set?

**Review by tay10r — CHANGES_REQUESTED** [review]

File: src/mixer.c [hunk 1]
```diff
@@ -1058,11 +1085,6 @@ int mixer_ctl_set_value(struct mixer_ctl *ctl, unsigned int id, int value)
         break;
 
     case SNDRV_CTL_ELEM_TYPE_INTEGER:
-        if ((value < mixer_ctl_get_range_min(ctl)) ||
```
[the remark this answers is not in the document]
Comment (dvdli) [reply src/mixer.c:1061]: I found that there are some drivers set the maximum of the control as UINT_MAX and the minimum as 0, but the max/min are stored in signed integers. Therefore, the range is 0 -> -1. 

I think not checking values in tinyalsa and checking these in drivers are better.

[the remark this answers is not in the document]
Comment (tay10r) [reply src/mixer.c:1061]: I see. I wonder how this affects user interfaces that use the range min and max to map the input range. In a spin box for example, or a slider bar. Not that this is an issue above, but I wonder if that's a separate issue that needs to be addressed.

**Review by tay10r — APPROVED** [review]

File: src/mixer.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (dvdli) [reply src/mixer.c:1061]: In my opinion, users still can trust almost every range of mixer control and get the min and max by the APIs mixer_ctl_get_range_min and mixer_ctl_get_range_max. Therefore, for the cases like a spin box or a slider bar, they still can work. We just let the driver to take the responsibility to check the range.

Note: tinyalsa also doesn't check the range in the `mixer_ctl_set_array`

**Pull request merged**